python/ray/tune/integration/ray_train.py

```diff
@@ -13,6 +14,12 @@
 class TuneReportCallback(UserCallback):
     """Propagate metrics and checkpoint paths from Ray Train workers to Ray Tune."""
 
+    def __init__(self):
+        assert _in_tune_session(), "TuneReportCallback must be used in a Tune session."
```

Comment:
raise a RuntimeError. assertion I mainly use for guarding types in internal code rather than user error.